src/vs/workbench/contrib/extensions/browser/extensionsActions.ts

```diff
@@ -329,45 +334,63 @@ export abstract class AbstractInstallAction extends ExtensionAction {
 
 		if (this.extension.deprecationInfo) {
```

Comment:
@sandy081 fyi please check, this one is a bit more complex adoption of the new dialog api

Reply:
Adoption is good, but I think it can be simplified further by running the action based on the choice output.

Reply:
Yes feel free to now chime in and improve further, I wanted to keep the change rather small.